http/src/lib.rs

```diff
@@ -254,76 +236,22 @@ impl<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>> ServerBuilder<M, S> {
 			extractor: self.meta_extractor,
 		};
 
-		let (l, srv) = hyper::Server::http(addr)?.handle(move |control| {
-			let hosts = hosts.lock().clone();
-			ServerHandler::new(
-				jsonrpc_handler.clone(),
-				cors_domains.clone(),
-				hosts,
-				request_middleware.clone(),
-				control,
-			)
-		})?;
-
-		// Add current host to allowed headers.
-		// NOTE: we need to use `l.addrs()` instead of `addr`
-		// it might be different!
-		{
-			let mut hosts = hosts_setter.lock();
-			if let Some(current_hosts) = hosts.take() {
-				let mut new_hosts = current_hosts.into_iter().collect::<HashSet<_>>();
-				for addr in l.addrs() {
-					let address = addr.to_string();
-					new_hosts.insert(address.clone().into());
-					new_hosts.insert(address.replace("127.0.0.1", "localhost").into());
-				}
-				// Override hosts
-				*hosts = Some(new_hosts.into_iter().collect());
-			}
-		}
+		let server_handler = ServerHandler {
```

Comment:
Taking `address` from running instance was quite useful if someone started the server with `port=0` - it causes random port to be allocated and that's why we were updating the allowed hosts header and storing it in the handle.